contracts/TetuVaultAssetManager.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.4;
+
+import "./openzeppelin/SafeERC20.sol";
+import "./openzeppelin/Math.sol";
+import "./third_party/balancer/IBVault.sol";
+import "./interfaces/ISmartVault.sol";
+import "./interfaces/IGauge.sol";
+import "./AssetManagerBase.sol";
+
+/// @title TetuVaultAssetManager
+/// @dev TetuVaultAssetManager can invest funds to the TETU vault.
+/// Rewards will be claimed and distributed by governance
+
+contract TetuVaultAssetManager is AssetManagerBase {
+  using SafeERC20 for IERC20;
+
+  // ***************************************************
+  //                VARIABLES
+  // ***************************************************
+
+  address public immutable tetuVault;
+
+  // ***************************************************
+  //                  EVENTS
+  // ***************************************************
+
+  event Invested(uint256 amount);
+  event Devested(uint256 amount);
+
+  // ***************************************************
+  //                CONSTRUCTOR
+  // ***************************************************
+
+  constructor(
+    IBVault balancerVault_,
+    address tetuVault_,
+    address underlying_
+  ) AssetManagerBase(balancerVault_, IERC20(underlying_)) {
+    require(tetuVault_ != address(0), "zero tetu vault");
+    tetuVault = tetuVault_;
+    IERC20(underlying_).safeIncreaseAllowance(tetuVault_, type(uint256).max);
+  }
+
+  // ***************************************************
+  //                VIEWS
+  // ***************************************************
+
+  /**
+   * @dev Checks balance of managed assets
+   */
+  function _getAUM() internal view override returns (uint256) {
+    return ISmartVault(tetuVault).underlyingBalanceInVault();
```

Comment:
underlyingBalanceWithInvestment is total balance

Reply:
are you sure that you need total balance? maybe what owned by this contract?

Reply:
Correct. Fixed